src/components/LogView/LogLineText.js

```diff
@@ -16,11 +16,51 @@ type Props = {
   port: ?string,
   text: string,
   highlightText: string[],
-  handleDoubleClick: ()=> void,
+  handleDoubleClick: () => void,
+  prettyPrint: boolean,
+}
+
+export function findJSONObjectsInLine(text: string) {
+  var startIndex = 0;
+  var numBraces = 0;
+  var chunks = [];
+  for (var i = 0; i < text.length; i++) {
+    if (text[i] === '{') {
+      if (numBraces === 0 && i !== 0) {
+        const lineBreak = (startIndex === 0) ? '' : '\n';
+        chunks.push(lineBreak + text.substring(startIndex, i));
+        startIndex = i;
+      }
+      numBraces++;
+    } else if (text[i] === '}') {
+      numBraces--;
+      if (numBraces === 0) {
+        try {
+          const lineBreak = (startIndex === 0) ? '' : '\n';
+          const jsonObj = JSON.parse(text.substring(startIndex, i + 1));
+          const formattedString = lineBreak + JSON.stringify(jsonObj, null, 2).replace(/"([^"]+)":/g, '$1:');
+          chunks.push(formattedString);
+          startIndex = i + 1;
+          console.log('this time it works!')
+        } catch (e) {
+          console.log('this always happens on first attempt to parse, for some reason');
```

Comment:
This is the weird thing. I noticed that it only happens when I run in the browser, when I run` npm test` I do not see this line printed at all.

Reply:
The implementation of JSON.parse is different between the browser, and in npm test (node.js), so it's not all that surprising. They are supposed to be functionally the same though. In this catch block, I would actually keep the text for the JSON as is, because it's possible that we logged some curly braces that aren't actually json